### DISCUSSION TITLE
Is StringValues supported for InlineData?

### DISCUSSION THREAD

**Discussion by jackwagons** [xunit/xunit#2326] [Question] [answered]
While attempting to use `Microsoft.Extensions.Primitives.StringValues` as one of the parameters for `InlineData` with `Theory`, I am getting `No test matches the given testcase filter...`.  

I verified this by simply changing the type to `System.String` and tests are discovered as expected.  Does `xunit` support `StringValues`?

Version Info:
- visual studio 2019
- xunit v2.4.0
- xunit.runner.visualstudio v2.4.00

**Comment by bradwilson** [comment, marked as answer]
There is no explicit support for `StringValues` in theory data, so my guess is that what you're running into is that `StringValues` isn't one of the things we know how to serialize in v2, and therefore your theory can't show all the pre-enumerated value sets inside Visual Studio (which in turn would also affect what things the filters can hit).

I'm converting this into a issue to ensure that `StringValues` is supported in v3.

**Reply by bradwilson** [reply]
So, I poked around a bit to see how `StringValues` would be treated in v3.

First: This test does pass in v3:

```csharp
public static TheoryData<StringValues> StringValuesData = new TheoryData<StringValues> { new StringValues(new[] { "hello", "world" }) };

[Theory]
[MemberData(nameof(StringValuesData))]
public static void TestingStringValues(StringValues data)
{
	Assert.Collection(
		data,
		d => Assert.Equal("hello", d),
		d => Assert.Equal("world", d)
	);
}
```

So `StringValues` is definitely _legal_ for theory data.

Second: `StringValues` is definitely not serializable. That means, when runners try to pre-enumerate the theory data (like the Visual Studio plugin does), it will fail to serialize (even in v3), which means the runner inside Visual Studio will only show this test method as a single test, regardless of the number of rows of data that are available. That also means you cannot filter for a specific row of data, either, since the system cannot see the rows of data.

I'm going to mark as closed, because I have verified that the feature does work, although your filter may not.
